proxy/src/transparency/server.rs

```diff
@@ -64,13 +66,15 @@ where
     ) -> Self {
         let recv_body_svc = HttpBodyNewSvc::new(stack.clone());
         let tcp = tcp::Proxy::new(tcp_connect_timeout, sensors.clone(), &executor);
+        let peek_timeout = Duration::from_millis(2);
```

Comment:
We may want this to be configurable, and/or different depending on inbound versus outbound?

Reply:
Setting this value too low means that we could lose HTTP telemetry on inbound requests that have non-negligible connect latency.  Setting it too high is also bad because this latency will be added to every mySQL connect.

I think we should set this to a value where HTTP connection from within the cluster will never time out and acknowledge that we may need to do something different for ingress in the future.

And yeah, configurable sounds good.